### PR TITLE
Start with no file even if some options have been specified

### LINKED ISSUE #1150: File is still required when passing one option to rizin
### Work environment

| Questions                                            | Answers
|------------------------------------------------------|--------------------
| OS/arch/bits (mandatory)                             | Debian arm 64, Ubuntu x86 32
| File format of the file you reverse (mandatory)      | PE, ELF etc.
| Architecture/bits of the file (mandatory)            | PPC, x86/32, x86/64 etc.
| `rizin -v` full output, **not truncated** (mandatory)  | rizin 0.2.0-git @ linux-x86-64, package: 0.2.0.1 (ret2libc) commit: c875be9afde5a6eed037249854b8a8759517263c, build: 2021-01-26__18:17:13

### Expected behavior
Rizin opens without any file opened and `asm.arch` set to `x86`.

### Actual behavior
```
Missing file to open
```

### Steps to reproduce the behavior
```
rizin -a x86
```

### Additional Logs, screenshots, source code,  configuration dump, ...

PR #930 was supposed to allow running `rizin` without any file, but that works only when there is no argument at all. It should be possible to open rizin without any file even when passing other arguments to `rizin`.

### PR DESCRIPTION
**Pull request by ret2libc**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

See attached issue for more details. In short, `rizin -a arm` should set asm.arch=arm and open no file.
This is not a great fix to be honest... The whole rizin file is very messy and probably would need a good refactoring to make sure everything is consistent.

**Test plan**

```
    rizin
    rizin --
    rizin -test
    rizin -- -test
    rizin -d -- -test
    rizin =
    rizin - < test/bins/elf/ls
    rizin -a arm
```

**Closing issues**

Closes https://github.com/rizinorg/rizin/issues/1150

### CODE REVIEW

File: librz/main/rizin.c [hunk 1]
```diff
@@ -911,7 +909,7 @@ RZ_API int rz_main_rizin(int argc, const char **argv) {
 			LISTS_FREE();
 			return 1;
 		}
-	} else if ((argc >= 2 && strcmp(argv[opt.ind - 1], "--")) || ((!strcmp(argv[opt.ind - 1], "--") && argv[opt.ind]))) {
+	} else if ((argc >= 2 && argv[opt.ind] && strcmp(argv[opt.ind], "--")) || ((!strcmp(argv[opt.ind - 1], "--") && argv[opt.ind]))) {
```
Comment (XVilka) [inline librz/main/rizin.c]: Can you extract this whole "if" in a small inline function?

**Review by XVilka — CHANGES_REQUESTED** [review]

File: librz/main/rizin.c (hunk 1 above)
Comment (wargio) [reply librz/main/rizin.c]: wait, didn't we change `-` to `=` ?

Comment (ret2libc) [reply librz/main/rizin.c]: yes we did. This is `--`.

For example, `rizin -- -test`, `rizin -d -- /bin/ls -l /`.

**Review by XVilka — APPROVED** [review]

**Comment by XVilka** [comment]
```
[XX] db/tools/rz rizin empty -R
RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc '!rizin -R ""' =
-- stderr
--- expected
+++ actual
@@ -1,1 +1,0 @@
-Missing file to open
```

**Comment by ret2libc** [comment]
> ```
> [XX] db/tools/rz rizin empty -R
> RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc '!rizin -R ""' =
> -- stderr
> --- expected
> +++ actual
> @@ -1,1 +1,0 @@
> -Missing file to open
> ```

I think this is exactly what this bug is fixing. I mean, `rizin` works without any file. So why `rizin -R ""` should not work? Note that `-R`accepts an argument, so `rizin -R ""` is passing an empty argument to `-R`. WDYT?

**Comment by XVilka** [comment]
Apparently not: https://travis-ci.com/github/rizinorg/rizin/jobs/515985001#L3679

**Comment by XVilka** [comment]
Yes, new behavior is better.

**Review by XVilka — APPROVED** [review]

**Pull request merged**